### PR TITLE
Add to docs based on feedback and fix for latest dev Dymos

### PR DESCRIPTION
**Pull request by johnjasa**
### Summary

This PR includes:
- minor changes suggested by users for clarity in the docs
- fixes for the latest dev version of Dymos

This PR has logic that checks if a user is using the latest version of Dymos and changes behavior regarding controls naming accordingly. After a Dymos release we can change this behavior so it just uses the correct syntax as detailed in #357

### Related Issues

- Resolves #

### Backwards incompatibilities

None

### New Dependencies

None

### CODE REVIEW

File: aviary/docs/getting_started/onboarding_level1.ipynb
[the remark this answers is not in the document]
Comment (jkirk5) [reply aviary/docs/getting_started/onboarding_level1.ipynb:1]: "associated report files" is a little vague, so perhaps some clarification could be added. "OpenMDAO and Dymos reports, such as optimization history, are saved there. Aviary's outputs and reports are placed in a folder named 'report', which is also created at that location." (?)

File: aviary/docs/user_guide/drawing_and_running_simple_missions.ipynb
Comment (jkirk5) [inline aviary/docs/user_guide/drawing_and_running_simple_missions.ipynb:1]: Update this text with any changes to previous section I commented on!

**Review by jkirk5 — APPROVED** [review]

File: aviary/docs/examples/level_2_detailed_takeoff_and_landing.ipynb [hunk 1]
```diff
@@ -45,6 +45,12 @@
     "However, you can easily combine these phases into a comprehensive mission by adding the takeoff, flight, and landing phases to the mission in that order.\n",
     "The `2DOF` mission method (not the `solved_2DOF` mission method used here) models the entire mission including takeoff and landing phases.\n",
     "\n",
+    "```{note}\n",
+    "The integration variable we use in the 2DOF model is distance, not time for all of the phases except groundroll.\n",
```
Comment (crecine) [inline aviary/docs/examples/level_2_detailed_takeoff_and_landing.ipynb:49]: for 2dof I'm seeing that the "time" variable in `_get_phase` is time for ascent, mass for cruise, time for descent, and it looks like all other phases use time as well. Should this be `solved_2DOF` here?

File: aviary/docs/getting_started/onboarding_level1.ipynb
[the remark this answers is not in the document]
Comment (johnjasa) [reply aviary/docs/getting_started/onboarding_level1.ipynb:1]: Thanks Jason for the suggestions! I've updated the verbiage in both places to:

When you run Aviary all of the outputs will be saved in the location where you run the command or run script.
Outputs files such as optimization history and Dymos output files are saved at that location.
A folder named 'report' will also be created there, and contains all of the associated report files for your runs.
These reports include [optimization and trajectory reports generated by OpenMDAO and Dymos](https://openmdao.org/newdocs/versions/latest/features/reports/reports_system.html).

File: aviary/validation_cases/benchmark_tests/test_FLOPS_based_sizing_N3CC.py [hunk 2]
```diff
@@ -37,7 +37,16 @@
     pyoptsparse = None
 
 
+from dymos.transcriptions.transcription_base import TranscriptionBase
+transcription_base = TranscriptionBase
+if hasattr(transcription_base, 'setup_polynomial_controls'):
```
[the remark this answers is not in the document]
Comment (johnjasa) [reply aviary/validation_cases/benchmark_tests/test_FLOPS_based_sizing_N3CC.py]: Good point, changed in three places

File: aviary/docs/examples/level_2_detailed_takeoff_and_landing.ipynb (hunk 1 above)
Comment (johnjasa) [reply aviary/docs/examples/level_2_detailed_takeoff_and_landing.ipynb]: Good catch, changed to `solved_2DOF`

**Review by crecine — APPROVED** [review]

**Pull request merged**